sled-agent/src/bootstrap/early_networking.rs

```diff
@@ -568,25 +569,67 @@ fn retry_policy_switch_mapping() -> ExponentialBackoff {
 /// The fields in this structure are those from
 /// [`super::params::RackInitializeRequest`] necessary for use beyond RSS. This
 /// is just for the initial rack configuration and cold boot purposes. Updates
-/// will come from Nexus in the future.
+/// come from Nexus.
 #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
 pub struct EarlyNetworkConfig {
-    // The version of data.
+    // The current generation number of data as stored in CRDB.
+    // The initial generation is set during RSS time and then only mutated
+    // by Nexus.
     pub generation: u64,
 
+    // Which version of the data structure do we have. This is to help with
+    // deserialization and conversion in future updates.
+    pub schema_version: u32,
+
+    // The actual configuration details
+    pub body: EarlyNetworkConfigBody,
+}
+
+/// This is the actual configuration of EarlyNetworking.
+///
+/// We nest it below the "header" of `generation` and `schema_version` so that
+/// we can perform partial deserialization of `EarlyNetworkConfig` to only read
+/// the header and defer deserialization of the body once we know the schema
+/// version. This is possible via the use of [`serde_json::value::RawValue`] in
+/// future (post-v1) deserialization paths.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct EarlyNetworkConfigBody {
     pub rack_subnet: Ipv6Addr,
 
     /// The external NTP server addresses.
     pub ntp_servers: Vec<String>,
 
-    /// A copy of the initial rack network configuration when we are in
-    /// generation `1`.
+    // Rack network configuration as delivered from RSS or Nexus
     pub rack_network_config: Option<RackNetworkConfig>,
 }
 
+// The first production version of the EarlyNetworkConfig.
+//
+// If this version is in the bootstore than we need to convert it to
+// `EarlyNetworkConfig`. We intend to convert this data on
```

Comment:
on what? :)

Reply:
Line removed in [a992afa](https://github.com/oxidecomputer/omicron/pull/4285/commits/a992afac9c5cb9622da4b8368a91d810cad58cc9)